### PR TITLE
Enable compiling with Visual Studio 2019 v142

### PR DESCRIPTION
**Pull request by malcolmdavey**
Not sure if this is the best approach, but Visual Studio 2019 (v142 toolset) complains about local class definitions which have pure virtual methods
- compile switch to give std behaviour for __cplusplus macro
- re-disable check which is not working for VS
- disabled some constexpr that isn't working for windows.

### CODE REVIEW

**Comment by malcolmdavey** [comment]
Fixing tests which have this issue:
https://github.com/eranpeer/FakeIt/issues/259

File: include/fakeit/StubbingProgress.hpp [hunk 1]
```diff
@@ -373,7 +373,7 @@ namespace fakeit {
             template <typename ...T, int ...N>
             static void CheckPositions(const std::tuple<ArgValue<T, N>...> arg_vals)
             {
-#if __cplusplus >= 201402L
+#if __cplusplus >= 201402L && !defined(_WIN32)
```
Comment (malcolmdavey) [inline include/fakeit/StubbingProgress.hpp:376]: Currently doesn't work for windows at all

File: tests/all_tests.vcxproj [hunk 2]
```diff
@@ -120,6 +120,12 @@
       <OptimizeReferences>true</OptimizeReferences>
     </Link>
   </ItemDefinitionGroup>
+  <ItemDefinitionGroup>
+    <ClCompile Condition="'$(PlatformToolset)' != 'v140'">
+      <AdditionalOptions>/Zc:__cplusplus</AdditionalOptions>
```
Comment (malcolmdavey) [inline tests/all_tests.vcxproj:125]: This forces VC++ to correctly define `__cplusplus`. It's good to test that this works correctly.

**Comment by FranckRJ** [comment]
Thanks for the fixes on VS, I usually only test with GCC / Clang (I don't really have a VS setup).

**Pull request merged**